tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+
+    Subclasses should provide a .workloads member to populate the 'wloads' field
+    of the experiments_conf for the Executor. Furthermore, subclasses may
+    overwrite the .hp_stress member containing the configuration used while
+    generating the random hotplug stress sequence. 
+
+    The test will be successful if the system does not halt or kill the
+    workloads because of hotplug stress. It means that the test passes if the
+    workload completes.
+    """
+    
+    test_conf = {
+        "ftrace" : {
+            "events" : [
+                "sched_overutilized",
+                "sched_energy_diff",
+                "sched_load_avg_task",
+                "sched_load_avg_cpu",
+                "sched_migrate_task",
+                "sched_switch",
+                "cpu_frequency",
+                "cpu_idle",
+                "cpu_capacity",
+                "cpuhp_enter",
+                "cpuhp_exit",
+                "cpuhp_multi_enter",
+            ],
+        },
+        "modules": ["cgroups", "hotplug"],
+    }
+
+    hp_stress = {
+        'seed' : None,                  # Seed of the random number generator
+        'sequence_len' : 100,           # Number of operations in the sequence
+        'sleep' : {
+            'min_ms' : 10,              # Min sleep duration between hotplugs 
+            'max_ms' : 100,             # Max sleep duration between hotplugs
+        },                              #   max_ms <= 0 will encode 'no sleep'
+        'max_cpus_off' : sys.maxint,    # Max number of CPUs plugged-off
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_CpuHotplugTest, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        conf = {
+            'tag' : 'energy_aware',
+            'flags' : ['ftrace', 'freeze_userspace'],
+            'sched_features' : 'ENERGY_AWARE',
+        }
+
+        return {
+            'wloads' : cls.workloads,
+            'confs' : [conf],
+        }
+
+    @classmethod
+    def _experimentsInit(cls, *args, **kwargs):
+        # calibration shoud be done before starting spurious hotplugs
+        if not cls.te._calib:
+            cls.te.calibration()
+        
+        # Choose a random seed explicitly if not given
+        if 'seed' not in cls.hp_stress or cls.hp_stress['seed'] is None:
+            random.seed() 
+            cls.hp_stress['seed'] = random.randint(0, sys.maxint)
+        random.seed(cls.hp_stress['seed'])
+        cls.te._log.info('Random sequence of CPU Hotplug generated with: ')
+        cls.te._log.info(cls.hp_stress)
+        with open(os.path.join(cls.te.res_dir, 'hotplug_cfg.json'), 'w') as f:
+            json.dump(cls.hp_stress, f, sort_keys=True, indent=4)
+       
+        # Play with (online) hotpluggable CPUs only
+        cls.hotpluggable_cpus = filter(
+                lambda x: cls.target.file_exists(hp._cpu_path(cls.target, x)),
```

Comment:
Yes that's what I had originally (a duplicate of `_cpu_path`), but it just felt cleaner not to re-do something that's already in devlib. The `HotplugModule.base_path` option is probably better though, I'll do the change